pkg/detector/detector.go

```diff
@@ -1063,7 +1064,11 @@ func (d *ResourceDetector) ReconcileResourceBinding(key util.QueueKey) error {
 		return nil
 	}
 
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
+	if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I'm sorry, i didn't describe it clearly, we can do an update first and then get in one action, just as:
https://github.com/karmada-io/karmada/blob/361978a3bf339db62b27757eacb45cf8194135ac/pkg/scheduler/scheduler.go#L820-L836

Reply:
Thanks for pointing out, didn't expect this method